imap_processing/ultra/l0/decom_ultra.py

```diff
@@ -344,3 +344,38 @@ def process_ultra_energy_spectra(ds: xr.Dataset) -> xr.Dataset:
     )
 
     return ds
+
+
+def process_ultra_macros_checksum(ds: xr.Dataset) -> xr.Dataset:
+    """
+    Unpack and decode Ultra MACROS CHECKSUM packets.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        Dataset containing macro checksums.
+
+    Returns
+    -------
+    dataset : xarray.Dataset
+        Dataset with unpacked and decoded checksum values.
+    """
+    checksums = []
+    fill = np.iinfo(np.uint16).max
+
+    for checksum in ds["checksums"]:
+        checksum_bytes = checksum.item()
+        arr = np.frombuffer(checksum_bytes, dtype=">u2")
+        checksums.append(arr)
```

Comment:
Rather than appending to a list here can you initialize an empty array above with the proper shape?

```python
nepochs = len(ds["epoch"])
nchecksums = 256
checksum_array = np.full((nepochs, nchecksums), fill, dtype=packed_dtype)
for i, checksum_row in enumerate(ds["checksums"]):
    checksum_array[i, :] = np.frombuffer(checksum_row.item(), dtype=packed_dtype)
```